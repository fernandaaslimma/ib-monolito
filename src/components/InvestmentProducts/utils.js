import { translate } from "../../utils/i18n";

export const verifyDays = (day, type) => {
  if (day == 0) {
    return translate("SAME_DAY_FUNDS");
  } else if (day == 1) {
    return type === "Útil"
      ? translate("ONE_WORKING_DAY_FUNDS")
      : translate("ONE_CONSECUTIVE_DAY_FUNDS");
  } else if (day > 1) {
    return type === "Útil"
      ? `${day}${translate("WORKING_DAYS_FUNDS")}`
      : `${day}${translate("CONSECUTIVE_DAYS_FUNDS")}`;
  }
};
