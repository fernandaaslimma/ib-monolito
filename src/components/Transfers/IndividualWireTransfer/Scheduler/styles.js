import styled, { css } from "styled-components";
import { rem, remFontSize } from "../../../../styles/tools";
import { negative300 } from "../../../../styles/settings";
import { RadioTag } from "../../../common/Radio/styles";

export const RadioWrapper = styled.div`
  ${RadioTag} {
    margin-left: ${rem(0)};
  }
  padding-left: ${props => (props.padding ? rem(props.padding) : 0)};
  padding-right: ${props => (props.padding ? rem(props.padding) : 0)};
  margin: 0 ${rem(8)} 0 ${rem(8)};
  display: flex;
  flex-direction: column;
`;

export const InputWrapper = styled.div`
  margin: ${rem(7)} ${rem(0)} ${rem(8)} ${rem(0)};

  ${({ margin }) =>
    margin &&
    css`
      margin: ${rem(margin.t ? margin.t : 7)} ${rem(margin.r ? margin.r : 0)}
        ${rem(margin.b ? margin.b : 8)} ${rem(margin.l ? margin.l : 0)};
    `}
`;

export const AlertMessageWrapper = styled.div`
  margin-top: ${rem(24)};

  ${({ noMargin }) =>
    noMargin &&
    css`
      margin-top: ${rem(0)};
    `}
`;

export const InputErrorMessage = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: 500;
  font-size: ${remFontSize(12)};
  line-height: 125%;
  letter-spacing: ${rem(0.45)};

  color: ${negative300};
`;

export const FavoredInfo = styled.div`
  margin: 0 ${rem(8)} 0 ${rem(8)};
`;
