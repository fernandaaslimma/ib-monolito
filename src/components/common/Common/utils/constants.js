export const PT_BR = "pt-BR";
export const EN_US = "en-US";
export const ZH_CN = "zh-CN";
export const SCREEN_LG = 1199;
export const SCREEN_MD = 992;
export const SCREEN_SM = 767;
export const SCREEN_XS = 449;
export const SCREEN_XXS = 319;
export const UP = "up";
export const DOWN = "down";
export const RIGHT = "right";
export const PAGINATION_TOTAL = "X-Total-Count";
export const DEFAULT_MOBILE_HEADER_HEIGHT = 60;
export const ASCENDING = "ASC";
export const DESCENDING = "DESC";
