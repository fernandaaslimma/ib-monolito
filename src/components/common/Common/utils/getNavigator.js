export const isMsBrowser = () => {
  if (typeof window !== "undefined") {
    return !!(navigator && navigator.msSaveOrOpenBlob);
  }
};

export const isMsEdge = () => {
  if (typeof window !== "undefined") {
    return !!(
      navigator &&
      navigator.msSaveOrOpenBlob &&
      navigator.appVersion.includes("Edge")
    );
  }
};

export const isInternetExplorer = () => isMsBrowser() && !isMsEdge();
