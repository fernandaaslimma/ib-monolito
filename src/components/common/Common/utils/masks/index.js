export const cpfMask = () => {
  return [
    /\d/,
    /\d/,
    /\d/,
    ".",
    /\d/,
    /\d/,
    /\d/,
    ".",
    /\d/,
    /\d/,
    /\d/,
    "-",
    /\d/,
    /\d/
  ];
};

export const cnpjMask = () => {
  return [
    /\d/,
    /\d/,
    ".",
    /\d/,
    /\d/,
    /\d/,
    ".",
    /\d/,
    /\d/,
    /\d/,
    "/",
    /\d/,
    /\d/,
    /\d/,
    /\d/,
    "-",
    /\d/,
    /\d/
  ];
};

export const cepMask = () => {
  return [/\d/, /\d/, /\d/, /\d/, /\d/, "-", /\d/, /\d/, /\d/];
};

export const datetimeMask = [
  /\d/,
  /\d/,
  "/",
  /\d/,
  /\d/,
  "/",
  /\d/,
  /\d/,
  /\d/,
  /\d/
];

export const maxValueMonth = [
  31,
  31,
  29,
  31,
  30,
  31,
  30,
  31,
  31,
  30,
  31,
  30,
  31
];

export const telephoneMask = [
  "(",
  /\d|\*/,
  /\d|\*/,
  ")",
  " ",
  /\d|\*/,
  /\d|\*/,
  /\d|\*/,
  /\d|\*/,
  "-",
  /\d|\*/,
  /\d|\*/,
  /\d|\*/,
  /\d|\*/,
  /\d|\*/
];

export const cellphoneMask = [
  "(",
  /\d|\*/,
  /\d|\*/,
  ")",
  " ",
  /\d|\*/,
  /\d|\*/,
  /\d|\*/,
  /\d|\*/,
  /\d|\*/,
  "-",
  /\d|\*/,
  /\d|\*/,
  /\d|\*/,
  /\d|\*/
];

export function telMask(value) {
  const raw = value.toString().replace(/\D/g, "");
  if (raw.length > 10) {
    return cellphoneMask;
  } else {
    return telephoneMask;
  }
}

export const formatOrder = ["yyyy", "yy", "mm", "dd", "HH", "MM", "SS"];
export const correctedDatePipe = (
  dateFormat,
  { minYear = 1, maxYear = 9999 } = {}
) => {
  const dateFormatArray = dateFormat
    .split(/[^dmyHMS]+/)
    .sort((a, b) => formatOrder.indexOf(a) - formatOrder.indexOf(b));
  return function(conformedValue) {
    const indexesOfPipedChars = [];
    const maxValue = {
      dd: 31,
      mm: 12,
      yy: 99,
      yyyy: maxYear,
      HH: 23,
      MM: 59,
      SS: 59
    };
    const minValue = {
      dd: 1,
      mm: 1,
      yy: 0,
      yyyy: minYear,
      HH: 0,
      MM: 0,
      SS: 0
    };
    const conformedValueArr = conformedValue.split("");

    // Check first digit
    dateFormatArray.forEach(format => {
      const position = dateFormat.indexOf(format);
      const maxFirstDigit = parseInt(
        maxValue[format].toString().substr(0, 1),
        10
      );

      if (parseInt(conformedValueArr[position], 10) > maxFirstDigit) {
        conformedValueArr[position + 1] = conformedValueArr[position];
        conformedValueArr[position] = 0;
        indexesOfPipedChars.push(position);
      }
    });

    // Check for invalid date
    let month = 0;
    const isInvalid = dateFormatArray.some(format => {
      const position = dateFormat.indexOf(format);
      const length = format.length;
      const textValue = conformedValue
        .substr(position, length)
        .replace(/\D/g, "");
      const value = parseInt(textValue, 10);
      if (format === "mm") {
        month = value || 0;
      }
      const maxValueForFormat =
        format === "dd" ? maxValueMonth[month] : maxValue[format];
      if (format === "yyyy" && (minYear !== 1 || maxYear !== 9999)) {
        const scopedMaxValue = parseInt(
          maxValue[format].toString().substring(0, textValue.length),
          10
        );
        const scopedMinValue = parseInt(
          minValue[format].toString().substring(0, textValue.length),
          10
        );
        return value < scopedMinValue || value > scopedMaxValue;
      }
      return (
        value > maxValueForFormat ||
        (textValue.length === length && value < minValue[format])
      );
    });

    if (isInvalid) {
      return false;
    }

    return {
      value: conformedValueArr.join(""),
      indexesOfPipedChars
    };
  };
};
