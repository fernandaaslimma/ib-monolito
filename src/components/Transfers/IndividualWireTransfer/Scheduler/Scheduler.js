import React, { Fragment, useContext, useState } from "react";
import moment from "moment";
import { InstanceContext } from "../IndividualWireTransfer";
import {
  InputWrapper,
  AlertMessageWrapper,
  InputErrorMessage,
  FavoredInfo
} from "./styles";
export const rem = value => `${value / 16}rem`;
import { AccTitle, FormWrapper } from "../styles";
import { AlertMessage, Input } from "react-bocombbm-components";
import {
  translate,
  getLanguage,
  getDateFieldPlaceholderByLocale,
  isPtBR
} from "../../../../utils/i18n";
import { isToday, checkDate } from "../../../../utils/validations/EFT";
import { modelBrazilianTime } from "../../../../utils/formatDate";
import {
  PT_BR_24H_FORMAT,
  EN_US_AM_PM_FORMAT,
  WIRETRANSFER_SERVICE,
  BRASILIA_UTC_OFFSET
} from "../../../../utils/constants";
import { blue, gray100, white } from "../../../../styles/settings";
import Buttons from "../../../common/Buttons";

function Scheduler({ title, hideAlert, hideSelection }) {
  const {
    props: {
      nextAvailableDate,
      availableDateRanges,
      transferData,
      loading,
      handleUserInputTranferDataDate,
      getAvailableDateRanges,
      serverTime
    },
    state: { isScheduled, inputedDate, validAsScheduled, commonValidToMoveOn },
    setStateValue,
    changeTransferDate,
    changeCommonValidDate,
    checkAvailabilitySchedule
  } = useContext(InstanceContext);

  const [canDoToday, setCanDoToday] = useState(true);
  const [untouched, setUntouched] = useState(true);
  const [focused, setFocused] = useState(false);
  const [availableRange, setAvailableRange] = useState({
    maxEndTime: null,
    minStartTime: null
  });

  const transferTimeSituations = {
    BEFORE_TIME: 0,
    ON_TIME: 1,
    AFTER_TIME: 2
  };
  const [transferTimeStatus, setTransferTimeStatus] = useState(
    transferTimeSituations.ON_TIME
  );

  React.useEffect(() => {
    const today = moment(serverTime)
      .utcOffset(BRASILIA_UTC_OFFSET)
      .format(getDateFieldPlaceholderByLocale());

    if (today !== nextAvailableDate) setCanDoToday(false);
    if (transferData.date !== nextAvailableDate)
      setStateValue("isScheduled", true);

    const now = Date.parse(
      moment(serverTime)
        .utcOffset(BRASILIA_UTC_OFFSET)
        .format()
    );

    const maxEndTime = Math.max(...availableDateRanges.map(x => x.endTime));
    const minStartTime = Math.min(...availableDateRanges.map(x => x.startTime));

    setAvailableRange({ maxEndTime, minStartTime });

    const minute = 60000;

    if (now <= maxEndTime - minute && now >= minStartTime - minute)
      setTransferTimeStatus(transferTimeSituations.ON_TIME);

    if (now <= minStartTime - minute)
      setTransferTimeStatus(transferTimeSituations.BEFORE_TIME);

    if (now >= maxEndTime - minute)
      setTransferTimeStatus(transferTimeSituations.AFTER_TIME);

    if (
      isToday(transferData.date, serverTime) &&
      !isScheduled &&
      transferTimeStatus.AFTER_TIME
    ) {
      setCanDoToday(false);
      setStateValue("isScheduled", true);
    }

    availableDateRanges.length
      ? setStateValue("validAsScheduled", true)
      : setStateValue("validAsScheduled", false);

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [transferData.date, availableDateRanges]);

  const checkDateInput = () =>
    inputedDate.length !== 0 && checkDate(inputedDate, true, serverTime);

  const enableScheduling = () => {
    setStateValue("isScheduled", true);
    changeTransferDate({ target: { value: "" } });
  };

  const enableToday = async () => {
    setStateValue("isScheduled", false);

    const today = moment(serverTime)
      .utcOffset(BRASILIA_UTC_OFFSET)
      .format(getDateFieldPlaceholderByLocale());

    handleUserInputTranferDataDate(today);

    await getAvailableDateRanges(WIRETRANSFER_SERVICE);

    changeCommonValidDate(true);
  };

  const getInvalidDateMessage = () => {
    switch (true) {
      case checkDateInput() && !validAsScheduled:
        return (
          <InputErrorMessage data-test="InvalidDay">
            {translate("NOT_BUSINESS_DAY")}
          </InputErrorMessage>
        );
      case checkDateInput() && isToday(inputedDate, serverTime):
        return (
          <InputErrorMessage>
            {translate("SELECT_TODAY_FOR_CURRENT_DATE")}
          </InputErrorMessage>
        );
      case !checkDateInput():
        return (
          <InputErrorMessage data-test="ProvideValidDate">
            {translate("PROVIDE_VALID_DATE")}
          </InputErrorMessage>
        );
    }
  };

  const getAlertMessage = () => {
    const rangeEnd = isPtBR()
      ? modelBrazilianTime(
          moment(availableRange.maxEndTime)
            .utcOffset(BRASILIA_UTC_OFFSET)
            .format(PT_BR_24H_FORMAT)
        )
      : moment(availableRange.maxEndTime)
          .utcOffset(BRASILIA_UTC_OFFSET)
          .format(EN_US_AM_PM_FORMAT);
    const rangeStart = isPtBR()
      ? modelBrazilianTime(
          moment(availableRange.minStartTime)
            .utcOffset(BRASILIA_UTC_OFFSET)
            .format(PT_BR_24H_FORMAT)
        )
      : moment(availableRange.minStartTime)
          .utcOffset(BRASILIA_UTC_OFFSET)
          .format(EN_US_AM_PM_FORMAT);

    const buildMessage = currentMessage => (
      <AlertMessageWrapper noMargin={hideSelection}>
        <AlertMessage
          dataTest="TransactionPeriod"
          icon="Attention"
          type="neutral"
          spacing={{
            top: "xs",
            bottom: "l",
            right: "xxs",
            left: "xxs"
          }}
        >
          {currentMessage}
        </AlertMessage>
      </AlertMessageWrapper>
    );

    let message = null;

    if (canDoToday && transferTimeStatus === transferTimeSituations.BEFORE_TIME)
      message = `${translate("INVALID_RANGE_1")} ${rangeStart} ${translate(
        "INVALID_RANGE_2"
      )} ${rangeEnd}${translate("INVALID_RANGE_3")} ${rangeStart}`;
    else
      message = `${translate(
        "VALIDATION_HOUR_SCHEDULE_1"
      )} ${rangeStart} ${translate("VALIDATION_HOUR_SCHEDULE_2")} ${rangeEnd}`;

    return buildMessage(message);
  };

  const dateInputChangeCallBack = event => {
    changeTransferDate(event);
    const value = event.target.value;

    if (event.target.value.indexOf("_") === -1) {
      const today = moment(serverTime)
        .utcOffset(BRASILIA_UTC_OFFSET)
        .startOf("day");
      const inputDate = moment(value, getDateFieldPlaceholderByLocale())
        .utcOffset(-3, true)
        .startOf("day");

      setStateValue("validAsScheduled", true);
      setFocused(false);
      inputDate.diff(today, "days") > 0 && checkAvailabilitySchedule();
    }
  };

  const checkIfIsValidDateAndEnableForward = () => {
    commonValidToMoveOn.date && setStateValue("disableForward", false);
  };

  const inputIsValid =
    checkDateInput() && !isToday(inputedDate, serverTime) && validAsScheduled;

  return (
    <Fragment>
      {title && <AccTitle>{title}</AccTitle>}
      <FormWrapper>
        {!hideSelection && (
          <FavoredInfo>
            <Buttons
              paddingWrapper={{ b: 6 }}
              buttons={[
                {
                  dataTest: "ButtonTransferToday",
                  children: translate("TODAY"),
                  onClick: () => enableToday(),
                  background: isScheduled ? `${white}` : `${blue}`,
                  color: isScheduled ? `${gray100}` : `${white}`,
                  border: `solid ${rem(1)} ${isScheduled ? gray100 : blue}`,
                  noHoverBackground: true,
                  fontSize: "14",
                  disabled: !canDoToday,
                  borderRadius: { rightTop: "0", rightBottom: "0" },
                  padding: { r: "20", l: "20" }
                },
                {
                  dataTest: "ButtonScheduledTransfer",
                  children: translate("SCHEDULE"),
                  onClick: () => enableScheduling(),
                  background: isScheduled ? `${blue}` : `${white}`,
                  color: isScheduled ? `${white}` : `${gray100}`,
                  border: `solid ${rem(1)} ${isScheduled ? blue : gray100}`,
                  noHoverBackground: true,
                  fontSize: "14",
                  borderRadius: { leftTop: "0", leftBottom: "0" },
                  padding: { r: "20", l: "20" }
                }
              ]}
            />
          </FavoredInfo>
        )}

        {isScheduled && !hideSelection && (
          <Fragment>
            <InputWrapper margin={{ t: 24, b: 24, l: 7, r: 0 }}>
              <Input
                onChange={e => dateInputChangeCallBack(e)}
                type="text"
                name="date"
                value={inputedDate}
                label={translate("DATE")}
                dataTest="transferDate"
                maskType="datetime"
                valid={inputIsValid}
                required
                locale={getLanguage()}
                onBlur={checkIfIsValidDateAndEnableForward}
                onFocus={() => {
                  setFocused(true);
                  setStateValue("disableForward", true);
                  setUntouched(false);
                }}
              />
            </InputWrapper>
            {!untouched && !focused && !loading && getInvalidDateMessage()}
          </Fragment>
        )}
        {!hideAlert && getAlertMessage()}
      </FormWrapper>
    </Fragment>
  );
}

export default Scheduler;
