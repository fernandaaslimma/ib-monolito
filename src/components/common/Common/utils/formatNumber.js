import { EN_US, PT_BR } from "../utils/constants";

export default function formatNumber(
  value,
  numberOptions = {},
  locale = EN_US
) {
  const validLocale =
    [EN_US, PT_BR].find(l => l.toLowerCase() === locale.toLowerCase()) || EN_US;

  const options = {};
  if (numberOptions.currency) {
    options.style = "currency";
    options.currency = numberOptions.currency;
  }

  if (numberOptions.digits) {
    options.minimumFractionDigits = numberOptions.digits;
    options.maximumFractionDigits = numberOptions.digits;
  }

  return new Intl.NumberFormat(validLocale, options).format(value);
}
