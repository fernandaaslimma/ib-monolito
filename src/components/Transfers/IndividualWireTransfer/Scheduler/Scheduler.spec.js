import React from "react";
import { shallow, mount } from "enzyme";
import deepClone from "../../../../utils/deepClone";
import Scheduler from "./Scheduler";
import { act } from "react-test-renderer";
import { isToday, checkDate } from "../../../../utils/validations/EFT";
import { isPtBR } from "../../../../utils/i18n";
import { Input } from "react-bocombbm-components";
import moment from "moment";
import { configure, fireEvent, render, screen } from "@testing-library/react";

configure({ testIdAttribute: "data-test" });
jest.mock("../../../../utils/validations/EFT");
jest.mock("../../../../utils/i18n");
jest.mock("uuid", () => {
  return { v4: () => "12345-abcvbcvb-23ddfsfd-4444" };
});

jest.mock("moment", () => jest.fn());
moment.mockImplementation(() => ({
  format: () => "hash",
  utcOffset: jest.fn().mockReturnThis(),
  startOf: () => ({ diff: () => 1 }),
  utc: jest.fn().mockReturnThis()
}));

const context = {
  props: {
    nextAvailableDate: "xpto",
    availableDateRanges: [
      { maxEndTime: 1605124811000, minStartTime: 1605103211 }
    ],
    transferData: { value: "5000,00", date: "" },
    loading: false,
    handleUserInputTranferDataDate: jest.fn(),
    getAvailableDateRanges: () => Promise.resolve()
  },
  state: {
    isScheduled: false,
    inputedDate: "11/20/2020",
    validAsScheduled: true
  },
  changeTransferDate: jest.fn(),
  checkAvailabilitySchedule: jest.fn(),
  setStateValue: jest.fn(),
  changeCommonValidDate: jest.fn()
};

describe("Scheduler component", () => {
  let setState, useStateSpy, useEffect;
  const mockUseEffect = () => {
    useEffect.mockImplementationOnce(f => f());
  };

  beforeEach(() => {
    setState = jest.fn();
    useStateSpy = jest.spyOn(React, "useState");
    useStateSpy.mockImplementation(init => [init, setState]);
    React.useContext = jest.fn(() => context);

    useEffect = jest.spyOn(React, "useEffect");
    mockUseEffect();

    isToday.mockImplementation(() => true);
    checkDate.mockImplementation(() => true);
    isPtBR.mockImplementation(() => true);
  });

  it.skip("Should match snapshot", () => {
    const component = shallow(<Scheduler />);
    expect(component).toMatchSnapshot();
  });

  it.skip("Should match snapshot not in pt-BR", () => {
    isPtBR.mockImplementation(() => false);
    const component = shallow(<Scheduler />);
    expect(component).toMatchSnapshot();
  });

  it.skip("Should match snapshot with scheduler and interact by focus", () => {
    const newContext = deepClone(context);
    newContext.state.isScheduled = true;
    newContext.state.validAsScheduled = true;
    isToday.mockImplementation(() => false);
    React.useContext = jest.fn(() => newContext);

    let component = mount(<Scheduler />);
    expect(component).toMatchSnapshot();

    act(() => {
      component.find(Input).prop("onFocus")();
    });
    component.update();
    expect(context.setStateValue).toHaveBeenCalled();
  });

  it.skip("should click on ButtonTransferToday", () => {
    render(<Scheduler />);
    const button = screen.getByTestId("ButtonTransferToday");
    fireEvent.click(button);
    expect(context.setStateValue).toHaveBeenCalled();
    expect(context.props.handleUserInputTranferDataDate).toHaveBeenCalled();
  });

  it.skip("should click on ButtonScheduledTransfer", () => {
    render(<Scheduler />);
    const button = screen.getByTestId("ButtonScheduledTransfer");
    fireEvent.click(button);
    expect(context.setStateValue).toHaveBeenCalled();
  });
});
