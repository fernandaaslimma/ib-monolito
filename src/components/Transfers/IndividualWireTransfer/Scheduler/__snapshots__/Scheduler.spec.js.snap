// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scheduler component Should match snapshot 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <Buttons
        buttons={
          Array [
            Object {
              "background": "#3175bd",
              "border": "solid 0.0625rem #3175bd",
              "borderRadius": Object {
                "rightBottom": "0",
                "rightTop": "0",
              },
              "children": undefined,
              "color": "#ffffff",
              "dataTest": "ButtonTransferToday",
              "disabled": false,
              "fontSize": "14",
              "noHoverBackground": true,
              "onClick": [Function],
              "padding": Object {
                "l": "20",
                "r": "20",
              },
            },
            Object {
              "background": "#ffffff",
              "border": "solid 0.0625rem #a4b2bb",
              "borderRadius": Object {
                "leftBottom": "0",
                "leftTop": "0",
              },
              "children": undefined,
              "color": "#a4b2bb",
              "dataTest": "ButtonScheduledTransfer",
              "fontSize": "14",
              "noHoverBackground": true,
              "onClick": [Function],
              "padding": Object {
                "l": "20",
                "r": "20",
              },
            },
          ]
        }
        paddingWrapper={
          Object {
            "b": 6,
          }
        }
      />
    </styled.div>
    <styled.div>
      <AlertMessage
        dataTest="TransactionPeriod"
        icon="Attention"
        spacing={
          Object {
            "bottom": "l",
            "left": "xxs",
            "right": "xxs",
            "top": "xs",
          }
        }
        type="neutral"
      >
        undefined hash undefined hash
      </AlertMessage>
    </styled.div>
  </styled.div>
</Fragment>
`;

exports[`Scheduler component Should match snapshot not in pt-BR 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <Buttons
        buttons={
          Array [
            Object {
              "background": "#3175bd",
              "border": "solid 0.0625rem #3175bd",
              "borderRadius": Object {
                "rightBottom": "0",
                "rightTop": "0",
              },
              "children": undefined,
              "color": "#ffffff",
              "dataTest": "ButtonTransferToday",
              "disabled": false,
              "fontSize": "14",
              "noHoverBackground": true,
              "onClick": [Function],
              "padding": Object {
                "l": "20",
                "r": "20",
              },
            },
            Object {
              "background": "#ffffff",
              "border": "solid 0.0625rem #a4b2bb",
              "borderRadius": Object {
                "leftBottom": "0",
                "leftTop": "0",
              },
              "children": undefined,
              "color": "#a4b2bb",
              "dataTest": "ButtonScheduledTransfer",
              "fontSize": "14",
              "noHoverBackground": true,
              "onClick": [Function],
              "padding": Object {
                "l": "20",
                "r": "20",
              },
            },
          ]
        }
        paddingWrapper={
          Object {
            "b": 6,
          }
        }
      />
    </styled.div>
    <styled.div>
      <AlertMessage
        dataTest="TransactionPeriod"
        icon="Attention"
        spacing={
          Object {
            "bottom": "l",
            "left": "xxs",
            "right": "xxs",
            "top": "xs",
          }
        }
        type="neutral"
      >
        undefined hash undefined hash
      </AlertMessage>
    </styled.div>
  </styled.div>
</Fragment>
`;

exports[`Scheduler component Should match snapshot with scheduler and interact by focus 1`] = `
<div
  className="sc-iqzUVk lhDyeD"
>
  <div
    className="sc-cfWELz khfrIN"
  >
    <div
      className="sc-eTpRJs jcNzOq"
    >
      <button
        className="sc-jTzLTM MPaOI"
        color="#a4b2bb"
        data-test="ButtonTransferToday"
        disabled={false}
        fontSize="14"
        height={null}
        onClick={[Function]}
        type="button"
        width={null}
      >
        <div
          className="sc-fjdhpX hAMeWv"
        >
          <div
            className="sc-jzJRlG fLqBJd"
          />
        </div>
      </button>
      <button
        className="sc-jTzLTM kpafqw"
        color="#ffffff"
        data-test="ButtonScheduledTransfer"
        disabled={false}
        fontSize="14"
        height={null}
        onClick={[Function]}
        type="button"
        width={null}
      >
        <div
          className="sc-fjdhpX hAMeWv"
        >
          <div
            className="sc-jzJRlG fLqBJd"
          />
        </div>
      </button>
    </div>
  </div>
  <div
    className="sc-gAmQfK bfJHUP"
  >
    <div
      className="sc-TOsTZ ftdOHJ"
    >
      <input
        className="sc-cHGsZl jqBJNN"
        data-test="transferDate"
        disabled={false}
        icon={null}
        label={null}
        name="date"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        required={true}
        type="text"
        value="11/20/2020"
      />
    </div>
  </div>
  <div
    className="sc-hAXbOi hgOTeD"
  >
    <div
      className="sc-fzsDOv igIEju"
      data-test="TransactionPeriod"
      spacing={
        Object {
          "bottom": "l",
          "left": "xxs",
          "right": "xxs",
          "top": "xs",
        }
      }
      type="neutral"
    >
      <span
        className="sc-iRbamj fimazz"
        color={null}
        height={24}
        spacing={
          Object {
            "left": "xxs",
            "right": "s",
          }
        }
        width={24}
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M0 0H24V24H0z"
            />
            <path
              d="M12 2.5c2.623 0 4.998 1.063 6.718 2.782C20.437 7.002 21.5 9.377 21.5 12s-1.063 4.998-2.782 6.718C16.998 20.437 14.623 21.5 12 21.5s-4.998-1.063-6.718-2.782C3.563 16.998 2.5 14.623 2.5 12s1.063-4.998 2.782-6.718C7.002 3.563 9.377 2.5 12 2.5zM12 4c-2.21 0-4.21.895-5.657 2.343C4.895 7.791 4 9.791 4 12c0 2.21.895 4.21 2.343 5.657C7.791 19.105 9.791 20 12 20c2.21 0 4.21-.895 5.657-2.343C19.105 16.209 20 14.209 20 12c0-2.21-.895-4.21-2.343-5.657C16.209 4.895 14.209 4 12 4zm0 10.5c.69 0 1.25.56 1.25 1.25S12.69 17 12 17s-1.25-.56-1.25-1.25.56-1.25 1.25-1.25zm.305-7.375c.552.01.992.466.982 1.018l-.074 4.25c-.01.551-.465.991-1.018.981-.552-.01-.992-.465-.982-1.017l.074-4.25c.01-.551.465-.991 1.018-.982z"
              fill="currentColor"
            />
          </g>
        </svg>
      </span>
      <p
        className="sc-gPWkxV kwAnYD"
      >
        undefined hash undefined hash
      </p>
    </div>
  </div>
</div>
`;
